com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs

```diff
@@ -608,86 +597,59 @@ void Swap(ref ScriptableRenderer r)
-                            colorLoadAction, RenderBufferStoreAction.Store, RenderBufferLoadAction.DontCare, RenderBufferStoreAction.DontCare);
-
-                        scaleBias = new Vector4(1, 1, 0, 0);;
-                        cmd.SetGlobalVector(ShaderPropertyId.scaleBias, scaleBias);
-                        cmd.DrawProcedural(Matrix4x4.identity, m_BlitMaterial, 0, MeshTopology.Quads, 4, 1, null);
-                    }
-                }
-                else
+                    isRenderToBackBufferTarget = cameraTargetHandle == cameraData.xr.renderTarget;
 #endif
-                {
-                    cmd.SetRenderTarget(cameraTarget, colorLoadAction, RenderBufferStoreAction.Store, RenderBufferLoadAction.DontCare, RenderBufferStoreAction.DontCare);
-#pragma warning disable 0618 // Obsolete usage: RenderTargetIdentifiers required here because of use of RenderTexture cameraData.targetTexture which is not managed by RTHandles
-                    cameraData.renderer.ConfigureCameraTarget(cameraTarget, cameraTarget);
-#pragma warning restore 0618
-                    cmd.SetViewProjectionMatrices(Matrix4x4.identity, Matrix4x4.identity);
+                RenderingUtils.FinalBlit(cmd, cameraData, isRenderToBackBufferTarget, GetSource(), cameraTargetHandle, colorLoadAction, RenderBufferStoreAction.Store, m_Materials.uber, 0);
 
-                    if ((m_Destination.nameID == BuiltinRenderTextureType.CameraTarget && !m_UseSwapBuffer) || (m_ResolveToScreen && m_UseSwapBuffer))
-                        cmd.SetViewport(cameraData.pixelRect);
-
-                    cmd.DrawMesh(RenderingUtils.fullscreenMesh, Matrix4x4.identity, m_Materials.uber);
+                bool setViewProjectionMatrices = true;
+#if ENABLE_VR && ENABLE_XR_MODULE
+                if (cameraData.xr.enabled)
+                    setViewProjectionMatrices = false;
+#endif
 
-                    // TODO: Implement swapbuffer in 2DRenderer so we can remove this
-                    // For now, when render post-processing in the middle of the camera stack (not resolving to screen)
-                    // we do an extra blit to ping pong results back to color texture. In future we should allow a Swap of the current active color texture
-                    // in the pipeline to avoid this extra blit.
-                    if (!m_ResolveToScreen && !m_UseSwapBuffer)
-                    {
-                        cmd.SetGlobalTexture(ShaderPropertyId.sourceTex, cameraTarget);
-                        cmd.SetRenderTarget(m_Source, RenderBufferLoadAction.DontCare, RenderBufferStoreAction.Store, RenderBufferLoadAction.DontCare, RenderBufferStoreAction.DontCare);
-                        cmd.DrawMesh(RenderingUtils.fullscreenMesh, Matrix4x4.identity, m_BlitMaterial);
-                    }
+                if (setViewProjectionMatrices)
+                    cmd.SetViewProjectionMatrices(Matrix4x4.identity, Matrix4x4.identity);
```

Comment:
`setViewProjectionMatrices` can probably be removed now that we use Blitter

Reply:
Yes, I would think so. Blitter does not use the matrices.